pkg/maps/policymap/policymap.go

```diff
@@ -237,89 +254,93 @@ func (key *PolicyKey) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(key) }
 func (key *PolicyKey) NewValue() bpf.MapValue    { return &PolicyEntry{} }
 
 func (key *PolicyKey) String() string {
-
+	prefixLen := key.Prefixlen - GetStaticPrefixBits()
 	trafficDirectionString := (trafficdirection.TrafficDirection)(key.TrafficDirection).String()
-	if key.DestPort != 0 {
-		return fmt.Sprintf("%s: %d %d/%d", trafficDirectionString, key.Identity, byteorder.NetworkToHost16(key.DestPort), key.Nexthdr)
+	if prefixLen > 0 {
+		dport := key.GetDestPort()
+		if prefixLen == 24 {
+			return fmt.Sprintf("%s: %d %d/%d", trafficDirectionString, key.Identity, dport, key.Nexthdr)
+		} else {
+			return fmt.Sprintf("%s: %d %d/%d/%d", trafficDirectionString, key.Identity, dport, prefixLen, key.Nexthdr)
+		}
 	}
 	return fmt.Sprintf("%s: %d", trafficDirectionString, key.Identity)
 }
 
-// ToHost returns a copy of key with fields converted from network byte-order
-// to host-byte-order if necessary.
-func (key *PolicyKey) ToHost() PolicyKey {
-	if key == nil {
-		return PolicyKey{}
+// NewKey returns a PolicyKey representing the specified parameters in network
+// byte-order.
+func NewKey(id uint32, dport uint16, proto uint8, trafficDirection uint8) PolicyKey {
+	// For now prefix length is derived from the proto and dport values
+	// This will have to be exposed to the caller when port ranges are supported.
+	prefixLen := GetStaticPrefixBits()
+	if proto != 0 {
+		prefixLen += 8
```

Comment:
Can these be derived using `sizeof` instead of hard cording the sizes?

Reply:
Done.